src/NBitcoin/Consensus.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using NBitcoin.BouncyCastle.Math;
+
+namespace NBitcoin
+{
+    public enum BuriedDeployments
+    {
+        /// <summary>
+        /// Height in coinbase.
+        /// </summary>
+        BIP34,
+
+        /// <summary>
+        /// Height in OP_CLTV.
+        /// </summary>
+        BIP65,
+
+        /// <summary>
+        /// Strict DER signature.
+        /// </summary>
+        BIP66
+    }
+
+    public class Consensus
+    {
+        /// <summary>
+        /// An extension to <see cref="Consensus"/> to enable additional options to the consensus data.
+        /// </summary>
+        public class ConsensusOptions
+        {
+        }
+
+        public ConsensusOptions Options { get; set; }
+
+        public class BuriedDeploymentsArray
+        {
+            private readonly int[] heights;
+
+            public BuriedDeploymentsArray()
+            {
+                this.heights = new int[Enum.GetValues(typeof(BuriedDeployments)).Length];
+            }
+
+            public int this[BuriedDeployments index]
+            {
+                get => this.heights[(int)index];
+                set => this.heights[(int)index] = value;
+            }
+        }
+
+        public class BIP9DeploymentsArray
+        {
+            private readonly BIP9DeploymentsParameters[] parameters;
+
+            public BIP9DeploymentsArray()
+            {
+                this.parameters = new BIP9DeploymentsParameters[Enum.GetValues(typeof(BIP9Deployments)).Length];
+            }
+
+            public BIP9DeploymentsParameters this[BIP9Deployments index]
+            {
+                get => this.parameters[(int)index];
+                set => this.parameters[(int)index] = value;
+            }
+        }
+
+        public Consensus()
+        {
+            this.BuriedDeployments = new BuriedDeploymentsArray();
+            this.BIP9Deployments = new BIP9DeploymentsArray();
+
+            this.ConsensusFactory = new ConsensusFactory()
+            {
+                Consensus = this
+            };
+        }
+
+        public BuriedDeploymentsArray BuriedDeployments { get; set; }
+
+        public BIP9DeploymentsArray BIP9Deployments { get; set; }
+
+        public int SubsidyHalvingInterval { get; set; }
+
+        public int MajorityEnforceBlockUpgrade { get; set; }
+
+        public int MajorityRejectBlockOutdated { get; set; }
+
+        public int MajorityWindow { get; set; }
+
+        public uint256 BIP34Hash { get; set; }
+
+        public Target PowLimit { get; set; }
+
+        public TimeSpan PowTargetTimespan { get; set; }
+
+        public TimeSpan PowTargetSpacing { get; set; }
+
+        public bool PowAllowMinDifficultyBlocks { get; set; }
+
+        public bool PowNoRetargeting { get; set; }
+
+        public uint256 HashGenesisBlock { get; set; }
+
+        /// <summary> The minimum amount of work the best chain should have. </summary>
+        public uint256 MinimumChainWork { get; set; }
```

Comment:
I've noticed that it's used only in IBD- if consensus chainwork is less- we are in IBD. 
What is the reason to have such a parameter? Is it useful?

Reply:
I guess we could take it out, it was here since forever (no harm though)